src/repl.coffee

```diff
@@ -7,7 +7,8 @@ CoffeeScript = require './coffeescript'
 
 replDefaults =
   prompt: 'coffee> ',
-  historyFile: path.join process.env.HOME, '.coffee_history' if process.env.HOME
+  historyFile: do (p = process.env.XDG_CACHE_HOME or process.env.HOME) ->
+    path.join p, '.coffee_history' if p
   historyMaxInputSize: 10240
```

Comment:
Further along in the file, a history file is created only if `opts.historyFile` is `true`. It's perhaps not ideal that two separate pieces of code are dealing with this, and unsetting `$HOME` is a slightly weird way to convey "don't write a history file", but I err on the side of making as small a change as possible.

```coffeescript
addHistory repl, opts.historyFile, opts.historyMaxInputSize if opts.historyFile
```

In my environment I've tested all four situations, with both `$HOME` and `$XDG_CACHE_HOME` set and unset.